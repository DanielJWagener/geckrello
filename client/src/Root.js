// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
