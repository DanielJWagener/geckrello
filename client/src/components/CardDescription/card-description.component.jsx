import React from "react";
import { connect } from "react-redux";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAlignLeft } from "@fortawesome/free-solid-svg-icons";

import { updateCardDescription } from "../../redux/cards/cards.actions";

import "./card-description.styles.scss";

export class CardDescription extends React.Component {
  state = {
    descriptionInput: "",
    mode: "prompt"
  };

  componentDidMount() {
    this.setState({ descriptionInput: this.props.card.description });
  }

  toggleMode = () => {
    if (this.state.mode === "prompt") {
      this.setState({ mode: "input" });
    } else if (this.state.mode === "input") {
      this.setState({ mode: "prompt" });
    }
  };

  onFormSubmit = e => {
    e.preventDefault();

    this.props.updateCardDescription(
      this.props.cardId,
      this.state.descriptionInput
    );
    this.setState({ mode: "prompt" });
  };

  handleChange = e => {
    this.setState({ descriptionInput: e.target.value });
  };

  descriptionExists = () =>
    this.state.descriptionInput && this.state.descriptionInput !== "\n";

  renderDescription = () =>
    this.descriptionExists() ? (
      <div className="description__display">{this.props.card.description}</div>
    ) : (
      <button className="description__prompt" onClick={this.toggleMode}>
        Add a description...
      </button>
    );

  renderEditButton = () =>
    this.descriptionExists() && this.state.mode === "prompt" ? (
      <button onClick={this.toggleMode} className="widget-heading__button">
        Edit
      </button>
    ) : null;

  render() {
    const descriptionInputForm = (
      <form onSubmit={this.onFormSubmit} className="modal-form">
        <div className="modal-form__group">
          <textarea
            name=""
            id=""
            cols="30"
            rows="5"
            className="modal-form__textarea"
            placeholder="Add a description..."
            onChange={this.handleChange}
            value={this.state.descriptionInput}
          ></textarea>
        </div>
        <div className="modal-form__group">
          <input type="submit" className="modal-form__submit" value="Save" />
          <button className="modal-form__cancel" onClick={this.toggleMode}>
            Cancel
          </button>
        </div>
      </form>
    );

    return (
      <div className="description">
        <div className="widget-heading">
          <FontAwesomeIcon
            icon={faAlignLeft}
            className="widget-heading__icon"
          />
          <h3 className="widget-heading__text">Description</h3>
          {this.renderEditButton()}
        </div>
        {this.state.mode === "prompt"
          ? this.renderDescription()
          : descriptionInputForm}
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const { cardId } = ownProps;
  const allCards = state.cards;
  const card = state.cards.filter(card => card._id === cardId)[0];
  return { card, allCards };
};

export default connect(mapStateToProps, { updateCardDescription })(
  CardDescription
);
