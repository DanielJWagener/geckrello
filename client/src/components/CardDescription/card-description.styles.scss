@import "../../sass/abstracts/_variables.scss";

.description {
  width: 100%;
  margin-bottom: 4rem;
  text-align: left;

  &__prompt {
    display: block;
    background-color: $component-background;
    border: none;
    cursor: pointer;
    padding: 0.5rem;
    border-radius: 5px;
    height: 3rem;
    &:hover {
      background-color: $component-background-dark;
    }
  }
  &__textarea {
    display: block;
    background-color: white;
    width: 100%;
    height: 4rem;
    padding: 0.5rem;
    border-radius: 5px;
  }

  &__display {
    display: block;
    margin-left: 7%;
    font-size: 1.6rem;
  }
}
