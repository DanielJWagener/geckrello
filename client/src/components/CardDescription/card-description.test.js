import { shallow } from "enzyme";
import React from "react";
import CardDescription from "./card-description.component";
import { CardDescription as CardDescriptionPure } from "./card-description.component";
import Root from "../../Root";

it("renders CardDescription component", () => {
  expect(
    shallow(<CardDescriptionPure card={{ description: "" }} />).debug()
  ).toMatchSnapshot();
});
