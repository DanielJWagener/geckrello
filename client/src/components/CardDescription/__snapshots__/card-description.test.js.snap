// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardDescription component 1`] = `
"<div className=\\"description\\">
  <div className=\\"widget-heading\\">
    <FontAwesomeIcon icon={{...}} className=\\"widget-heading__icon\\" border={false} mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
    <h3 className=\\"widget-heading__text\\">
      Description
    </h3>
  </div>
  <button className=\\"description__prompt\\" onClick={[Function]}>
    Add a description...
  </button>
</div>"
`;
